app/src/app/[lng]/auth/signup/page.tsx

```diff
@@ -105,9 +90,9 @@ export default function Signup({
         setError(message);
         return;
       }
-
-      const callbackParam = callbackUrl ? `&callbackUrl=${callbackUrl}` : "";
-      const nextCallbackUrl = `/auth/check-email?email=${data.email}${callbackParam}`;
+      const queryParamsString = new URLSearchParams(queryParams).toString();
+      const callbackParam = callbackUrl ? "&" : "";
+      const nextCallbackUrl = `/auth/check-email?email_address=${data.email}&${callbackParam}${queryParamsString}`;
```

Comment:
This leads to a double & if there is a callback URL (since that's not appended to the & anymore in the previous line).